### DISCUSSION TITLE
How to understand the internal of IREE project?

### DISCUSSION THREAD

**Discussion by FlintWangacc** [iree-org/iree#17995] [General]
I am using ```iree-import-onnx``` to familiarize with IREE project.
I found out some features about IREE project.
1. ```iree-import-onnx``` is a python based neural network frontend which use pybind to interact with C++ libraries.
2. The IREE project dialects are a LLVM external project.

Actually I don't know how to get a big picture of IREE project. Is there any advices for novice developers？

**Comment by ScottTodd** [comment]
Check out the architecture overview diagram, both in the repository README: https://github.com/iree-org/iree?tab=readme-ov-file#architecture-overview and on the website: https://iree.dev/#project-architecture. The slides and recordings for presentations and talks may also be useful: https://github.com/iree-org/iree?tab=readme-ov-file#presentations-and-talks.

For `iree-import-onnx` in particular, I'd also look at https://iree.dev/guides/ml-frameworks/onnx/ and https://github.com/llvm/torch-mlir
